### PR TITLE
Abstracts bulk acceptance

### PR DESCRIPTION
**Pull request by mvidalgarcia**
image

### CODE REVIEW

File: indico/modules/events/abstracts/forms.py [hunk 1]
```diff
@@ -212,7 +212,11 @@ class AbstractJudgmentFormBase(IndicoForm):
     accepted_track = QuerySelectField(_("Track"), [HiddenUnless('judgment', AbstractAction.accept)],
                                       get_label='title', allow_blank=True, blank_text=_("Choose a track..."),
                                       description=_("The abstract will be accepted in this track"))
-    accepted_contrib_type = QuerySelectField(_("Contribution type"), [HiddenUnless('judgment', AbstractAction.accept)],
+    override_contrib_type = BooleanField(_('Override contribution type'),
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/forms.py]: You probably want to remove this one if there are no contrib types (see `__init__`)

Comment (nop33) [inline indico/modules/events/abstracts/forms.py]: We tend to use double quotes for translatable strings, but it's not really enforced, however endorsed!

File: indico/modules/events/abstracts/forms.py [hunk 2]
```diff
@@ -212,7 +212,11 @@ class AbstractJudgmentFormBase(IndicoForm):
     accepted_track = QuerySelectField(_("Track"), [HiddenUnless('judgment', AbstractAction.accept)],
                                       get_label='title', allow_blank=True, blank_text=_("Choose a track..."),
                                       description=_("The abstract will be accepted in this track"))
```
Comment (nop33) [inline indico/modules/events/abstracts/forms.py]: We should make sure that the description text of the fields is adjusted accordingly for the bulk operations (we are not talking about a single abstract anymore, but many).

Comment (ThiefMaster) [inline indico/modules/events/abstracts/forms.py]: Not related to this PR though ;)

Comment (nop33) [inline indico/modules/events/abstracts/forms.py]: The PR is named "Abstracts bulk acceptance", what do you mean it's not related? Yes, it's functional without it, but the description texts are misleading.

**Comment by ThiefMaster** [comment]
Just saying they were the same before and didn't change in this PR

**Comment by nop33** [comment]
And I am making sure we won't forget it before merging it so that the 
problem stops here ;)

File: indico/modules/events/abstracts/forms.py [hunk 3]
```diff
@@ -342,6 +349,12 @@ class BulkAbstractJudgmentForm(AbstractJudgmentFormBase):
     def __init__(self, *args, **kwargs):
         self.event = kwargs.pop('event')
         super(BulkAbstractJudgmentForm, self).__init__(*args, **kwargs)
+        self.accepted_track.description = _("The abstracts will be accepted in this track")
+        self.accepted_contrib_type.description = _("The abstracts will be converted into a contribution of this type")
+        self.session.description = _("The generated contributions will be allocated in this session")
+        self.duplicate_of.description = _("The selected abstracts will be marked as duplicate of this one")
+        self.merged_into.description = _("The selected abstracts will be merged onto this one")
+        self.merge_persons.description = ("Authors and speakers of the selected abstracts will be added to this one")
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/forms.py]: missing `_` in `_(...)`

File: indico/modules/events/abstracts/forms.py [hunk 4]
```diff
@@ -342,6 +349,12 @@ class BulkAbstractJudgmentForm(AbstractJudgmentFormBase):
     def __init__(self, *args, **kwargs):
         self.event = kwargs.pop('event')
         super(BulkAbstractJudgmentForm, self).__init__(*args, **kwargs)
+        self.accepted_track.description = _("The abstracts will be accepted in this track")
+        self.accepted_contrib_type.description = _("The abstracts will be converted into a contribution of this type")
+        self.session.description = _("The generated contributions will be allocated in this session")
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/forms.py]: `if self.session: ...`

Comment (ThiefMaster) [inline indico/modules/events/abstracts/forms.py]: Same with the two fields above as they can be removed too if there are no tracks/types

File: indico/modules/events/abstracts/forms.py [hunk 5]
```diff
@@ -212,7 +212,11 @@ class AbstractJudgmentFormBase(IndicoForm):
     accepted_track = QuerySelectField(_("Track"), [HiddenUnless('judgment', AbstractAction.accept)],
                                       get_label='title', allow_blank=True, blank_text=_("Choose a track..."),
                                       description=_("The abstract will be accepted in this track"))
-    accepted_contrib_type = QuerySelectField(_("Contribution type"), [HiddenUnless('judgment', AbstractAction.accept)],
+    override_contrib_type = BooleanField(_("Override contribution type"),
+                                         [HiddenUnless('judgment', AbstractAction.accept)], widget=SwitchWidget(),
+                                         description=_("Override the contribution type for all selected abstracts"))
+    accepted_contrib_type = QuerySelectField(_("Contribution type"), [HiddenUnless('judgment', AbstractAction.accept),
+                                                                      HiddenUnless('override_contrib_type')],
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/forms.py]: you cannot have more than one HiddenUnless on the same field

**Pull request merged**